### PR TITLE
add datafusion

### PR DESCRIPTION
**Pull request by waitingkuo**
Add datafusion
not yet find a "c6a.4xlarge, 500gb gp2" vm to test this

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Thank you! I'm very interested in this.

File: datafusion/benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Install Rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rust-init.sh
+bash rust-init.sh -y
+source .cargo/env
```
Comment (qoega) [inline datafusion/benchmark.sh:6]:
```suggestion
source ~/.cargo/env
```

**Comment by qoega** [comment]
I tried to run it on c6a.4xlarge 
```
[2.638, 0.331, 0.305],
[0.516, 0.362, 0.362],
[1.312, 0.938, 0.888],
[1.427, 0.483, 0.487],
[3.006, 2.958, 2.953],
[null, null, null],
[0.479, 0.358, 0.353],
[0.450, 0.362, 0.367],
[3.952, 3.568, 3.370],
[6.014, 5.381, 5.570],
[2.301, 1.797, 1.726],
[2.810, 2.234, 2.139],
[null, null, null],
[null, null, null],
[null, null, null],
[4.722, 4.707, 4.469],
[null, null, null],
[null, null, null],
[null, null, null],
[0.786, 0.510, 0.494],
[14.218, 8.460, 8.409],
[17.441, 11.445, 11.307],
[41.716, 28.732, 28.761],
[129.533, 95.898, 95.534],
[7.422, 5.185, 5.246],
[10.154, 9.899, 10.094],
[13.262, 11.534, 11.464],
[18.279, 13.104, 14.166],
[225.516, 232.958, 234.723],
[2.859, 3.152, 3.202],
[8.917, 6.679, 6.655],
[11.631, 7.802, 7.855],
[0.081, 0.054, 0.056],
[null, null, null],
[null, null, null],
[4.864, 4.564, 4.709],
[null, null, null],
[null, null, null],
[null, null, null],
[null, null, null],
[null, null, null],
[null, null, null],
[null, null, null],
```

**Comment by alexey-milovidov** [comment]
@waitingkuo Let's submit these results and then improve?

**Comment by waitingkuo** [comment]
@alexey-milovidov we haven't fixed some issues in our master branch, i'll modify the installation part to build the latest and then submit it

**Comment by waitingkuo** [comment]
@qoega  thank you!

**Comment by waitingkuo** [comment]
close apache/arrow-datafusion#2902 and apache/arrow-datafusion#3048

**Comment by waitingkuo** [comment]
@qoega thank you

unfortunately, i just found a issue here. i'll make it right again as soon as possible

**Comment by qoega** [comment]
No problem. I just have that instance ready for benchmarks and can pull and run anytime.

**Comment by waitingkuo** [comment]
@qoega  @alexey-milovidov i submitted an issues for parquet data #18 

`hits.parquet` and `hits_{n}.parquet` are slight different

**Comment by waitingkuo** [comment]
datafusion parquet importer doesn't support schema for now, it's inferred from parquet metadata directly.

**Comment by waitingkuo** [comment]
i did the benchmark again. it works now. I thought that there were some issues it turned out that i used hits_0.parquet to do the quick test. 
It's ready to be merged again :D

**Comment by waitingkuo** [comment]
@qoega  @alexey-milovidov i've made the result up to date. please let me if there's anything i need to improve. thanks~

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@waitingkuo Thank you!
I also wanted to re-run on AWS for better comparison and to check reproducibility, but let's firstly merge as is.

**Pull request merged**